chunk/inmemory_storage_client.go

```diff
@@ -125,7 +125,8 @@ func (m *MockStorage) BatchWrite(_ context.Context, batch WriteBatch) error {
 			items = append(items, mockItem{})
 			copy(items[i+1:], items[i:])
 		} else {
-			return fmt.Errorf("Dupe write")
+			// Ignore duplicate write
+			continue
```

Comment:
The reason duplicate writes are now appearing is because of the metric name `IndexEntry`. Multiple chunks have the same `bucketHashKey` and the `RangeValue` doesn't include the `chunkID`. 

I will add back the error and dedupe the writes.